src/python/pants/base/payload_field.py

```diff
@@ -139,8 +139,12 @@ def _compute_fingerprint(self):
     hasher.update(self._rel_path)
     for source in sorted(self.relative_to_buildroot()):
       hasher.update(source)
-      with open(os.path.join(get_buildroot(), source), 'rb') as f:
-        hasher.update(f.read())
+      abs_path = os.path.join(get_buildroot(), source)
+      if not os.path.exists(abs_path):
+        hasher.update('\a\n\n\t[FILE DOES NOT EXIST]\t\n\n\a')
+      else:
```

Comment:
I had to add this because sources in java_wire_libraries that come from an unpacked_jar won't actually exist until after the unpacked_jar step runs. Obviously this raises concerns over doing cache invalidation properly; I think in the wire case these are taken care of by the dependency on the jar it's unpacking (because if the jar file changes, the wire_library should get invalidated transitively), but we might not want this code here because we probably want to catch missing sources in other cases. Maybe I need to make a new payload field that subclasses sources, eg OptionalSourcesField?